src/Transform/ONNX/InstrumentONNXPass.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------- InstrumentONNXPass.cpp - Instrumentation ---------------------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file implements a Function level pass that inserts instrumentation
+// for ONNX ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include <set>
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Interfaces/CallInterfaces.h"
+#include "mlir/Pass/Pass.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/Interface/ShapeInferenceOpInterface.hpp"
+#include "src/Pass/Passes.hpp"
+
+using namespace mlir;
+
+namespace {
+
+/*!
+ * This pass insert KrnlInstrumentOp before and after each ONNX ops
+ */
+
+// TO FIX: this option should be put in the ONNXMlirOptions category
```

Comment:
I was actually just looking at this for some options that we'd like to add as well without adding the dependency of various libs on MainUtils. I think clang has a CommonOptions library which defines a variety of command line options that can then be re-used elsewhere by linking against this specific library which is just used for options. How do you feel about doing something similar here? Then we could add all of the options to a single library and link each of the onnx mlir libs that need to know about the options against that?

Reply:
@sstamenova that seems reasonable to me, @chentong319 what do you think?

Reply:
I just found out that mlir pass manager provides mechanism to define the option inside the pass. I should try that.

Reply:
@sstamenova @caoimhinuibrian  MLIR provides some example of defining options in a pass. It seems to me that options for each pass can be defined within the pass. In this way, options are closed to their consumer code. The patter used by clang, which put the option together in a separate file, is good for options shared by different passes, such as options for target architecture, optimization level, debugging status.  I will do further investigation.